babel/dates.py

```diff
@@ -1482,7 +1482,13 @@ def format_era(self, char: str, num: int) -> str:
     def format_year(self, char: str, num: int) -> str:
         value = self.value.year
         if char.isupper():
-            value = self.value.isocalendar()[0]
+            month = self.value.month
+            if month == 1:
+                if self.value.day < 7 and self.get_week_of_year() >= 52:
```

Comment:
Combine into one if block perhaps?
```suggestion
            if month == 1 and self.value.day < 7 and self.get_week_of_year() >= 52:
```

Reply:
It is an alternative to `match month case 1 ... case 12 ...`.